src/server/auth/index.js

```diff
@@ -0,0 +1,70 @@
+let _ = require('lodash');
+let Boom = require('boom');
+let Joi = require('joi');
+let hapiAuthCookie = require('hapi-auth-cookie');
+let strategyMap = require('./lib/authStrategyMap');
+
+module.exports = (kbnServer, server, config) => {
+  if (!config.get('auth.enabled')) return;
+  let name = config.get('auth.strategy');
+  let strategy = strategyMap.get(name);
+  if (strategy == null) throw new Error(`There is no registered authentication strategy with the name "${name}".`);
+  if (_.isFunction(strategy.init)) strategy.init(server);
+
+  server.register(hapiAuthCookie, (error) => {
+    if (error != null) throw error;
+
+    let options = {
+      cookie: 'sid',
+      password: config.get('auth.encryptionKey'),
+      ttl: config.get('auth.sessionTimeout'),
+      clearInvalid: true,
+      isHttpOnly: false,
+      keepAlive: true,
+      isSecure: true,
+      redirectTo: '/login'
+    };
+
+    if (_.isFunction(strategy.validate)) {
+      options.validateFunc = (request, session, callback) => strategy.validate(request, session).nodeify(callback);
```

Comment:
We can't use `.nodeify()` here, without first wrapping the promise in bluebird. It is quite likely that `strategy.validate` won't always return a bluebird promise. Perhaps something like this:

```js
let { attempt } = require('bluebird')
options.validateFunc = (request, session, callback) => {
  attempt(strategy.validate, [request, session], strategy).nodeify(callback);
}
```